gobot/handlers/pull_request_create.go

```diff
@@ -0,0 +1,483 @@
+	}
+
+	// Commit the changes with signature
+	signature := &object.Signature{
+		Name:  requestData.Name,
+		Email: requestData.Email,
+		When:  time.Now(),
+	}
+
+	commitMsg := fmt.Sprintf("%s: %s \n\n Signed-off-by: %s <%s>", SkillStr, requestData.Task_description, signature.Name, signature.Email)
+
+	commit, err := wt.Commit(commitMsg, &git.CommitOptions{
+		Author:    signature,
+		Committer: signature,
+	})
+	if err != nil {
+		return "", fmt.Errorf("Error committing skill related changes: %v", err)
+	}
+	return commit.String(), nil
+}
+
+func (prt *PullRequestTask) createKnowledgeCommit(wt *git.Worktree, requestData KnowledgePRRequest) (string, error) {
+	// Write the requestData to a file
+	workDir, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("Error getting current working directory: %v", err)
+	}
+
+	dirPath := path.Join(workDir, prt.branchName, TaxonomyPath, RepoKnowledgePath, prt.branchName)
+	err = os.MkdirAll(dirPath, os.ModePerm)
+	if err != nil {
+		return "", fmt.Errorf("Error creating directory: %v", err)
+	}
+
+	filePath := path.Join(dirPath, YamlFileName)
+	yamlFile, err := os.Create(filePath)
+	if err != nil {
+		return "", fmt.Errorf("Error creating knowledge yaml file: %v", err)
+	}
+	defer yamlFile.Close()
+
+	// Convert requestData to yaml
+	yamlData, err := prt.generateKnowledgeYaml(requestData)
+	if err != nil {
+		return "", fmt.Errorf("Error generating knowledge yaml from the pull request data: %v", err)
+	}
+
+	_, err = yamlFile.WriteString(yamlData)
+	if err != nil {
+		return "", fmt.Errorf("Error writing to yaml file: %v", err)
+	}
+
+	// Add the attribution file to the PR
+	attributionFilePath := path.Join(dirPath, AttributionFileName)
+	attributionFile, err := os.Create(attributionFilePath)
+	if err != nil {
+		return "", fmt.Errorf("Error creating knowledge attribution file: %v", err)
+	}
+	defer attributionFile.Close()
+
+	// Convert requestData to yaml
+	attributionFileData := prt.generateKnowledgeAttributionData(requestData)
+
+	_, err = attributionFile.WriteString(attributionFileData)
+	if err != nil {
+		return "", fmt.Errorf("Error writing to knowledge attribution file: %v", err)
+	}
+
+	skillDir := path.Join("."+RepoKnowledgePath, prt.branchName)
+	_, err = wt.Add(skillDir)
+	if err != nil {
+		return "", fmt.Errorf("Error adding knowledge files to git: %v", err)
+	}
+
+	// Commit the changes with signature
+	signature := &object.Signature{
+		Name:  requestData.Name,
+		Email: requestData.Email,
+		When:  time.Now(),
+	}
+
+	commitMsg := fmt.Sprintf("%s: %s \n\n Signed-off-by: %s <%s>", KnowledgeStr, requestData.Task_description, signature.Name, signature.Email)
+
+	commit, err := wt.Commit(commitMsg, &git.CommitOptions{
+		Author:    signature,
+		Committer: signature,
+	})
+	if err != nil {
+		return "", fmt.Errorf("Error committing knowledge related changes: %v", err)
+	}
+
+	return commit.String(), nil
+}
+
+func (prt *PullRequestTask) pushCommit(repo *git.Repository) error {
```

Comment:
Related to above, still https auth. Please confirm this is not an issue.

Reply:
With proper permissions it should work. Works fine with my local taxonomy fork, with write permission enabled.